### DISCUSSION TITLE
Show Grain Boundary in Abaqus Output

### DISCUSSION THREAD

**Discussion by lreigbua** [neperfepx/neper#421] [Q&A] [answered]
Hello,

I am trying to find a simple way to show the grain boundaries in an Abaqus output.

This is simple if you include cohesive elements or use discontinuous interfaces with the option -faset. But I am trying to do it with continuous interface, and the option -faset does not produce any interface surfaces.

->neper -M n30-id1.tess -interface discontinuous -faset internal -format inp

This command does it, but it separates the mesh.

->neper -M n30-id1.tess -interface continuous -faset internal -format inp

If you use continuous, the surfaces are not created anymore.

Am I missing something?
Is there any way to obtain a surface such as the one -faset gives you in the .inp between grains but with continuous interfaces?
Is there other way I am not thinking of?

Thanks!

**Comment by rquey** [comment]
It would be `-dim 2,3`, which writes both the 2D mesh (of the grain boundaries) and the 3D mesh. It also generates elsets named `face1`, `face2`, etc., which correspond to the individual grain boundaries.

**Reply by lreigbua** [reply]
Hello Romain,

First of all, thank you very much for the quick replay. Your software is being super helpful.

I have tried this, but I think having the 2d element sets does not solve the problem. I think that in order to show the cell boundaries Abaqus needs either a surface or a partition. It also creates the problem that the 2D elements require a separate material assignment.

I think this may have to be solved through Abaqus scripting. Is there any way for the mesh to yield the nodesets at each boundary between the cells, instead of the element sets?

If if I find a way I will post it here.

Regards and thanks again,
Luis

**Comment by lreigbua** [comment, marked as answer]
I found a way to do it in Abaqus without modifying anything in Neper. In the visualization tab, you have to create a display group for each element set. Then, when adding all the display groups to the display, it will show the feature edges for each element set (each grain/cell).

**Reply by Sin-Sal** [reply]
Hello,
I generated .inp file using neper and since it had errors while trying to import to Abaqus i tried -faset none and successfully it was imported to abaqus. but there is 1 thing that i can not understand, when I import it to Abaqus unlike the file from neper i can't see grains and grain boundaries that I generated in neper, it's just a simple cube with mesh and I don't know why and what to do.

**Reply by lreigbua** [reply]
Hello Sin-Sal,

This code helped me a lot in understanding how to bring Neper RVEs into Abaqus: https://github.com/DylanAgius/Neper2Abaqus
It is meant to be use with Huang's UMAT, but you could adapt it for other UMATs.

Let me know if you need more help.

**Reply by Sin-Sal** [reply]
Hello again, 
Thank you and yes I have already checked it but there are 2 issues:
1- about how to write the material priperties in the excel file it says that it is addressed in a file abd the link is provided but in that link I couldn't find the file so i don't know how to write material properties in the sheet. do you have this file?
2- i generated the neper file as it says but after importing that according to the instruction to the Abaqus it's a simple cube with hex mesh and no grains do you see grains when you import it? Maybe I'm making a mistake while editing the file (as it says i should write 2 things at the end of the input file in the end)
I have been stuck here since long ago and I appreciate if you help me.
